core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceTypeUtil.java

```diff
@@ -82,6 +99,73 @@
                     .put("0", false)
                     .build();
 
+    /**
+     * Try to get a public static method {@link MethodHandle} with a single {@link String} argument
+     * in clazz.
+     * @param <T> the expected result
+     * @param clazz the class being introspected
+     * @param methodName the method expected
+     * @return a function that takes a {@link String} and produces a {@link T}
+     */
+    @Nullable
+    private static <T> MethodHandle getPublicStaticMethodHandle(final Class<T> clazz,
+                                                                final String methodName) {
+        try {
+            final MethodType methodType = MethodType.methodType(clazz, String.class);
+            return MethodHandles.publicLookup().findStatic(clazz, methodName, methodType);
+        } catch (Throwable t) {
+            // No valid public static method found
+            return null;
+        }
```

Comment:
- Could we also double check the static method returns the object that's compatible with `clazz`? i.e.

      if (!clazz.isAssignableFrom(methodType.returnType())) {
          return;
      }
- Could you also add the test case for the static method that returns an incompatible type? e.g.

      public class Foo {
          public static Bar of(String value) { new Bar(value); }
      }
      public class Bar { ... }

Reply:
> * Could you also add the test case for the static method that returns an incompatible type? e.g.
>   ```
>   public class Foo {
>       public static Bar of(String value) { new Bar(value); }
>   }
>   public class Bar { ... }
>   ```

Excellent point. Done!